app/scripts/background.js

```diff
@@ -482,8 +483,15 @@ async function triggerUi() {
     tabs.length > 0 &&
     tabs[0].extData &&
     tabs[0].extData.indexOf('vivaldi_tab') > -1;
-  if ((isVivaldi || !popupIsOpen) && !currentlyActiveMetamaskTab) {
-    await notificationManager.showPopup();
+  if (!uiIsTriggering && (isVivaldi || !popupIsOpen) && !currentlyActiveMetamaskTab) {
+    uiIsTriggering = true;
+    try {
+      await notificationManager.showPopup();
+    } catch (err) {
+      log.error('Error triggering UI', err);
```

Comment:
Could you remove this case? I'd rather the error bubble up and give a more complete stack trace if this fails.

```suggestion
```

Reply:
Absolutely :) lmk if any other changes are needed.